File: laby/paradygmatySem3/laby8/main.scala
def printAll (args: Any*): Unit = {
    for (arg <- args) 
    {
        if (arg == null) print("Null\n")
        else print(arg.getClass.getSimpleName + ": " + arg + '\n')
    }
}

object main {
    def main(args: Array[String]): Unit = {
        printAll(1, 2, "aloha", Nil);
        printAll();
        printAll(null);
        printAll(null, "k");
  }
}

File: laby/paradygmatySem3/laby8/main.java
import java.util.ArrayList;

class Main {
	public static void main(String ...args) {
        printAll(1, 2, "aloha", new ArrayList<>());
        printAll();
        printAll(null);
        printAll(null, "k");
    }

    private static void printAll(Object ...args) {

        if (args == null) {
                System.out.println("Null");
                return;
        }

        for (Object arg : args) {
            if (arg == null) System.out.print("Null\n");
            else System.out.print(arg.getClass().getSimpleName() + ": " + arg + '\n');
        }
    }
}